### PR TITLE
[PD-3369] refactor analytics reducer

### PR DESCRIPTION
**Pull request by mwarnsleyDE**
## What's this PR do?
Refactors the analytics reducers

## Screenshots (if appropriate)
N/A

## Where should the reviewer start?
Start inside of the analytics application

## How should this be manually tested?
Pull down the repository. Go into analytics and make sure all of the actions associated with clicking on anything still works

## Any background context you want to provide?
More code based than visual based so looking at the code would be better

### CODE REVIEW

File: gulp/src/analytics/components/Table/TableSearch.js [hunk 1]
```diff
@@ -54,6 +53,4 @@ TableSearch.propTypes = {
   dispatch: PropTypes.func.isRequired,
 };
 
-export default connect(state => ({
-  analytics: state.pageLoadData,
-}))(TableSearch);
+export default connect(state => ({}))(TableSearch);
```
Comment (aljordan82) [inline gulp/src/analytics/components/Table/TableSearch.js]: Does this need to be connected? Why not just exporting TableSearch?

File: gulp/src/analytics/components/Tabs/TabsContainer.js [hunk 2]
```diff
@@ -43,29 +42,27 @@ class TabsContainer extends Component {
   }
 
   render() {
-    const {analytics, tabsMenuActive, closeMenus, dispatch} = this.props;
+    const {navigation, tabsMenuActive, closeMenus, dispatch, printing, stateCustomRange} = this.props;
```
Comment (aljordan82) [inline gulp/src/analytics/components/Tabs/TabsContainer.js]: is 'printing' used?

**Comment by aljordan82** [comment]
ChartContainer.js implements `shouldComponentUpdate() `,it needs to be updated, also test Printing

File: gulp/src/analytics/components/Tabs/TabsContainer.js (hunk 2 above)
Comment (mwarnsleyDE) [inline gulp/src/analytics/components/Tabs/TabsContainer.js]: @aljordan82 printing is what I changed that you had in for the printing. It's the same thing as what you had just in a different format and coming from the store

File: gulp/src/analytics/components/Table/TableSearch.js (hunk 1 above)
Comment (mwarnsleyDE) [inline gulp/src/analytics/components/Table/TableSearch.js]: @aljordan82 I updated this and just passed the dispatch from the parent component

File: gulp/src/analytics/actions/table-filter-actions.js [hunk 3]
```diff
@@ -97,10 +97,10 @@ export function doActivateTimeline(active) {
   return async (dispatch, getState, {api, history}) => {
     try {
       dispatch(markNavLoadingAction(true));
-      const currentStartRange = getState().pageLoadData.globalStartDate;
-      const currentEndRange = getState().pageLoadData.globalEndDate;
-      const currentReport = getState().pageLoadData.activeReport;
-      const navigation = getState().pageLoadData.navigation;
+      const currentStartRange = getState().header.globalStartDate;
```
Comment (darrint) [inline gulp/src/analytics/actions/table-filter-actions.js]: :+1: I like seeing `pageLoadData` broken up and replaced by more meaningful things.

File: gulp/src/analytics/components/BreadCrumb.js [hunk 4]
```diff
@@ -21,8 +20,17 @@ class BreadCrumb extends Component {
...
+   */
   id: PropTypes.number,
+  /**
+   * Actual crumbs and their names being supplied
+   */
   crumbs: PropTypes.array.isRequired,
+  /**
+   * Action that takes place once you click on the breadcrumbs
+   */
   breadcrumbClick: PropTypes.func,
```
Comment (darrint) [inline gulp/src/analytics/components/BreadCrumb.js]: This should be named `onBreadcrumbClick`. I think it's ok to go without change this for now but in the future let's keep this in mind: https://github.com/DirectEmployers/MyJobs/blob/quality-control/doc/source/developers/styleguides/javascript.rst#react-conventions

**Pull request merged**